docs/glome-login.md

```diff
@@ -17,61 +17,49 @@ If the authorization code matches the one calculated internally by
 
+This document describes version 2 of GLOME Login, see also
+[RFD001](rfd/001.md) for a rationale of the protocol elements.
+
 ## Implementation
 
 The current version of the GLOME login protocol uses the
 [standard GLOME variant](protocol.md#variants).
 Counters are set to constant `0` since only a single set of messages
 is exchanged.
 
-*   GLOME handshake information and tags are encoded as Base64-encoded URLs or
-    "base64url"
-    [[RFC4648 section 5](https://tools.ietf.org/html/rfc4648#section-5)].
-*   Initial message from the GLOME login client to the server contains the
-    context required for authorization (i.e. host identity, requested action).
-*   The authorization context is sent in clear for easier debuggability and
-    reducing the likelihood of human errors (e.g. incomplete URL copy and
-    paste).
-*   Server's public key can be identified by:
-    *   7-bit service key identifier and message tag prefix (of any
-        length, including 0).
-    *   7-bit service key prefix and message tag prefix (of any length,
-        including 0),
-*   Using a message tag prefix provides an additional protection against channel
-    errors (e.g. caused by operator errors).
-*   The message sent from the GLOME login client to the server contains the context required for authorization (i.e. host identity, requested action).
-*   In this protocol the client and the server sign identical messages.
-    the client to the server, and therefore is omitted.
+The protocol assumes that the client (i.e., the Linux machine being accessed),
+knows the public key of the server. Required elements of the protocol are:
+
+* A host identifier that uniquely identifies the client.
+* An action on the client that needs to be authorized by the server.
+
+These optional elements can provide additional information to the server:
+
+* A host identifier type, among which the host identifier is unique.
+* A server key index, to tell the server which private key to use.
+* A message tag prefix, to allow error detection on the server side.
+
+The client combines these elements into a challenge string, which the server
+validates and responds to with a GLOME tag. GLOME Login challenges are suitable
+for embedding into a URL.
 
 ### Challenge request format
 
 The GLOME login client generates the challenge in the form:
 
-```
-v<V>/<glome-handshake>[/<message>]/
+```abnf
+challenge = "v2/" handshake-segment "/" message "/"
 
-glome-handshake := base64url(
-    <prefix-type>
-    <prefix7>
-    <eph-key>
-    [<prefixN>]
-  )
+handshake-segment = Base64_urlsafe( prefix client-public-key [message-tag-prefix] )
 
-message := [<hostid-type>:]<hostid>[/<action>]
+message = host-segment "/" action-segment
+host-segment = EscapePathSegment( [hostid-type ":"] hostid )
+action-segment = EscapePathSegment(action)
 ```
 
-where <fields> have the following meanings:
+The individual elements of this specification are described in the subsections below.
```

Comment:
I think I'd write "The individual elements of this specification and the encoding functions are described in the subsections below".